### PR TITLE
Improved code with better security, rate-limits and error handling

### PR DESCRIPTION
**Pull request by Codimow**

### CODE REVIEW

**Review by kiritocode1 — COMMENTED** [review]
LGTM , ship it :P

**Comment by kiritocode1** [comment]
image

**Review by kiritocode1 — APPROVED** [review]

**Review by kiritocode1 — COMMENTED** [review]
LOOKS FINE , improve

**Review by kiritocode1 — APPROVED** [review]

File: backend/index.ts [hunk 1]
```diff
@@ -1,33 +1,42 @@
...
+import express, { Request, Response, NextFunction } from "express";
 import cors from "cors";
 import cookieParser from "cookie-parser";
 import Router from "./routes/index";
 import { SERVER_PORT } from './config/index';
 
-// === 1 - CREATE SERVER ===
-const server = express();
+// Initialize the Express application
+const app = express();
```
Comment (sarvesh371) [inline backend/index.ts:8]: How does it help to change variable name from server to app?

File: backend/utils/postgres.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ const pool = new Pool({
   port: parseInt(DB_PORT || "5432"),
   ssl: SSL === "true" ? true : SSL === "false" ? false : undefined,
 });
-
+s
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:13]: Please remove this

File: backend/utils/postgres.ts [hunk 3]
```diff
@@ -19,11 +19,21 @@ export async function runQuery(queryText: string, values: any[] = []): Promise<a
...
+  try {
+    await pool.end();
+    console.log("Database connection pool closed.");
+  } catch (error) {
+    console.error("Error closing the database connection pool:", error);
+  }
+}
+
+// Function to get the current pool status (optional)
+export function getPoolStatus(): string {
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:37]: this function is not getting used anywhere in the code then why add?

File: backend/utils/postgres.ts [hunk 4]
```diff
@@ -19,11 +19,21 @@ export async function runQuery(queryText: string, values: any[] = []): Promise<a
     return res.rows;
   } catch (error) {
     console.error("Database query error:", error);
-    throw new Error(`Database query failed: ${error}`);
+    throw new Error(`Database query failed: ${error.message}`);
   }
 }
 
 // Clean up pool (if needed, e.g., on app shutdown)
 export async function closePool(): Promise<void> {
```
Comment (sarvesh371) [inline backend/utils/postgres.ts:27]: Same this function is not getting used anywhere, either remove or use it

File: discord_bot/src/commands/accessKey.js [hunk 5]
```diff
@@ -42,43 +36,34 @@ module.exports = {
...
             if (!roleId) {
                 return await interaction.reply({
-                    content: 'No role ID found for this access key.',
-                    ephemeral: true
-                });
+                    content: 'No role ID found for this access key.',ephemeral: true});
             }
 
-            console.log('Role ID to add:', roleId); // Add this line for debugging
+            console.log('Role ID to add:', roleId); // debugging
```
Comment (sarvesh371) [inline discord_bot/src/commands/accessKey.js:49]: How is this log helpful for us?

Comment (sarvesh371) [reply discord_bot/src/commands/accessKey.js:49]: unnecessary logs looks like

File: discord_bot/src/utils/supabase.js [hunk 6]
```diff
@@ -2,128 +2,130 @@ const { createClient } = require('@supabase/supabase-js');
 require('dotenv').config();
 const crypto = require('crypto');
 
-const supabaseUrl = process.env.SUPABASE_URL;
-const supabaseKey = process.env.SUPABASE_KEY;
-
-const supabase = createClient(supabaseUrl, supabaseKey);
+const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_KEY);
```
Comment (sarvesh371) [inline discord_bot/src/utils/supabase.js:5]: This should be in try catch?

File: backend/utils/postgres.ts (hunk 4 above)
Comment (Codimow) [reply backend/utils/postgres.ts:27]: there is a difference between them

File: discord_bot/src/commands/accessKey.js (hunk 5 above)
Comment (Codimow) [reply discord_bot/src/commands/accessKey.js:49]: nope it is useful in testing phase

File: discord_bot/src/utils/supabase.js (hunk 6 above)
Comment (Codimow) [reply discord_bot/src/utils/supabase.js:5]: just shorter

File: discord_bot/src/commands/accessKey.js (hunk 5 above)
Comment (sarvesh371) [reply discord_bot/src/commands/accessKey.js:49]: but should not be pushed to production, you can add in local

File: discord_bot/src/utils/supabase.js (hunk 6 above)
Comment (sarvesh371) [reply discord_bot/src/utils/supabase.js:5]: ???

File: backend/utils/postgres.ts (hunk 4 above)
Comment (sarvesh371) [reply backend/utils/postgres.ts:27]: Why have function when not used?

**Pull request closed without merging**